packages/script-csp/README.md

```diff
@@ -0,0 +1,70 @@
+# AMP-Toolbox Script CSP
+
+
+[![npm version](https://badge.fury.io/js/%40ampproject%2Ftoolbox-script-csp.svg)](https://badge.fury.io/js/%40ampproject%2Ftoolbox-script-csp)
+
+Calculates the [Content Security Policy (CSP)](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP)
+hash for the given script in the format expected by [`amp-script`](https://amp.dev/documentation/components/amp-script/).
+
+CSP is required when using `amp-script` with inline or cross-origin scripts.
+
+## Usage
+
+Install via:
+
+```
+$ npm install @ampproject/toolbox-script-csp
+```
+
+### Using the module
+
+The module exposes a single function, `calculateHash(src, options?)`.
+
+`src` can be either a string (in which case UTF8 encoding is assumed) or a
+[`DataView`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView)
+such as a [`Buffer`](https://nodejs.org/api/buffer.html).
+
+`options` is used to override the default options.
+
+#### Options
+
+*   `algorithm` overrides the hashing algorithm to use. Currently, the only
+    algorithm supported by `amp-script` is `sha384` which is the default value.
+
+### Example
```

Comment:
Can you mention how whitespace is handledl

Reply:
I think it's beyond scope? It just hashes the given input as-is, including whitespace. `amp-script` also expects the hash to include them and CSP works the same.

Reply:
Yes, but we should mention it:

```
Here is an example on how to use the `calculateHash` function to generate an `amp-script` CSP header. It's important to note that whitespace is critical when passing a script string to `calculateHash`.
```

Reply:
Done